### ISSUE TITLE
Should subpath mapping/pattern support conditions?

### ISSUE THREAD

**Issue by yyx990803** [lukeed/resolve.exports#1]
Example:

```json
{
  "exports": {
    ".": {
      "import": "./main.js",
      "require": "./cjs.js"
    },
    "./dir-mapped/": {
      "import": "./dir/",
      "require": "./dir-cjs/"
    }
  }
}
```

The [docs](https://nodejs.org/api/packages.html#packages_subpath_patterns) doesn't really specify this, but I think it makes sense. /cc @MylesBorins

**Comment by lukeed** [comment]
Oh interesting. I'd never seen this but agree that it makes sense.
(I'm assuming you're referring to `"./dir-mapped/"` specifically.)

Just confirmed that it works in Node.js too (12.18, 14.15 – but _not_ 12.14), so I think it was intended. 
If @MylesBorins can confirm that's officially supported, I'll go ahead and implement this to maintain parity.

**Comment by yyx990803** [comment]
This is the only blocker for Vite to switch to use this package - great job!

**Comment by MylesBorins** [comment]
Conditions were aways supported for subpath folders. v12.17.0 was the version of 12.x where we unflagged ESM, as such it can be expected that if you are using ESM unflagged today subpath folder mapping will be supported.

That said we have deprecated [subpath folder mapping](https://nodejs.org/dist/latest-v15.x/docs/api/packages.html#packages_subpath_folder_mappings) encouraging developers to instead use [subpath patterns](https://nodejs.org/dist/latest-v15.x/docs/api/packages.html#packages_subpath_patterns), which can do everything that subpath folder mapping can do and more.

Subpath patterns landed in v12.20.0 and v14.13.0.

**Comment by lukeed** [comment]
Thank you Myles :) Both are supported here, just to make it easy. The bits for subpath folder mapping is easy to remove at a later date.

With that confirmation, I went ahead and added this missing piece. Will publish a patch version now.
Thanks!
